### PR TITLE
chore(redis): add shellspec tests for redis6-sentinel-post-start.sh

### PR DESCRIPTION
**Pull request by weicao**
## Summary
- Make `redis6-sentinel-post-start.sh` shellspec-testable: change shebang to bash, add ut_mode guard, wrap in `acl_set_user_for_redis6_sentinel()`, add SOURCED guard
- Add 6 shellspec test cases covering: empty password no-op, successful ACL setup, unset/custom port, TLS flags, custom sentinel user

## Test plan
- [ ] CI shellspec passes
- [ ] CI shell-check passes

### CODE REVIEW

File: addons/redis/scripts/redis6-sentinel-post-start.sh [hunk 1]
```diff
@@ -1,11 +1,24 @@
-#!/bin/sh
+#!/bin/bash
 
-set -e
+# shellcheck disable=SC2034
+ut_mode="false"
+test || __() {
+  set -e;
```
Comment (leon-ape) [inline addons/redis/scripts/redis6-sentinel-post-start.sh:6]: `set -e` is inside `__()` and is not executed in normal production execution, so failures from `ACL SETUSER` or `ACL SAVE` can be masked and the postStart script can still return 0.

**Comment by weicao** [comment]
Superseded by #2853 (combined shellspec test PR per Leon's review guidance)
